src/js/client/editprefset-dialogs-widget.js

```diff
@@ -0,0 +1,304 @@
+/**
+ * Dialog boxes for the DevPTT.
+ */
+"use strict";
+
+var gpii  = fluid.registerNamespace("gpii");
+fluid.registerNamespace("gpii.devpmt");
+fluid.registerNamespace("gpii.devpmt.dialogs");
+
+/**
+ * Add NP Set Context Dialog
+ */
+fluid.defaults("gpii.devpmt.dialogs.addContextDialog", {
+    gradeNames: ["gpii.devpmt.dialogs.confirmDialog"],
+    templates: {
+        initial: "editprefset-addContext-dialog"
+    },
+    model: {
+        contextId: "",
+        contextToCopy: "",
+        contextNames: [],
+        fieldErrors: ""
+    },
+    selectors: {
+        contextIdInput: "#pmt-add-context-name-input",
+        contextToCopySelect: "#pmt-context-to-copy-select"
+    },
+    bindings: {
+        contextIdInput: "contextId",
+        contextToCopySelect: "contextToCopy"
+    },
+    invokers: {
+        acceptConfirmDialog: {
+            funcName: "gpii.devpmt.dialogs.addContextDialog.acceptConfirmDialog",
+            args: ["{that}", "{gpii.devpmt.editPrefs}", "{arguments}.0"]
+        }
+    }
+});
+
+/**
+ * Dialog for adding a new preference set (context) to a preferences safe (prefset).
+ * Includes a dropdown that can be used to select a context to copy the
+ * initial settings from.
+ *
+ * @param  {that} that           Add Context Dialog instance
+ * @param  {editPrefs} editPrefs Central gpii.devpmt.editPrefs component for the page.
+ */
+gpii.devpmt.dialogs.addContextDialog.acceptConfirmDialog = function (that, editPrefs, event) {
+    // We don't want the form to actually submit the page, just to enable
+    // activating the submit button on Enter
+    event.preventDefault();
+
+    // Validate context name, eventually this will be replaced with
+    // our new json schema work.
+    if (that.model.contextId === "") {
+        that.applier.change("fieldErrors", "Please enter a name.");
+        that.reRender();
```

Comment:
Does it make sense to add a model listener for `fieldErrors` to trigger `that.reRender()` whenever `fieldErrors` is filled up.

Reply:
This is a good idea. Updated.